### ISSUE TITLE
[Rocket.Chat+.App] Invalid url

### ISSUE THREAD

**Issue by netdevfighter** [RocketChat/Rocket.Chat#2677]
![rocketchat](https://cloud.githubusercontent.com/assets/4456399/14101600/cd788eae-f595-11e5-958c-a8b3155d79a3.png)

The URL pattern is not correct implemented.

Your Rocket.Chat version: 0.24.0
Client version: latest of Apple store

**Comment by engelgabriel** [comment]
@netdevfighter are you sure http://srv-rc-test.bs.kae.de.oneandone.net/ is accessible?
The message is not saying that the URL is bad formatted, but it is saying that it didn't find a valid Rocket.Chat instance.

@rodrigok maybe we can make the error message more clear? 
"No valid instance found at the url"
"INVALID RESPONCE"

**Comment by netdevfighter** [comment]
That's confused me ... 

My input was "https://srv-rc-test.bs.kae.de.oneandone.net" also with https. After the click on the connect button I get the url back without 's'.



Some more detials about this box:
´´´
tlindenmann@bluefire:/opt/rocketchat$ host srv-rc-test.bs.kae.de.oneandone.net
srv-rc-test.bs.kae.de.oneandone.net is an alias for et-0.srv-rc-test.bs.kae.de.oneandone.net.
et-0.srv-rc-test.bs.kae.de.oneandone.net has address 172.20.39.136
tlindenmann@bluefire:/opt/rocketchat$ nmap srv-rc-test.bs.kae.de.oneandone.net

Starting Nmap 6.40 ( http://nmap.org ) at 2016-03-29 18:12 CEST
Nmap scan report for srv-rc-test.bs.kae.de.oneandone.net (172.20.39.136)
Host is up (0.056s latency).
rDNS record for 172.20.39.136: et-0.srv-rc-test.bs.kae.de.oneandone.net
Not shown: 995 closed ports
PORT     STATE SERVICE
22/tcp   open  ssh
80/tcp   open  http
111/tcp  open  rpcbind
443/tcp  open  https
3000/tcp open  ppp

Nmap done: 1 IP address (1 host up) scanned in 0.81 seconds
``` 

Has the application problem with cname?

The website-url is configured to "https://srv-rc-test.bs.kae.de.oneandone.net"

**Comment by engelgabriel** [comment]
Is this a private network? I can access from here.

```
This site can’t be reached

srv-rc-test.bs.kae.de.oneandone.net’s server DNS address could not be found.
DNS_PROBE_FINISHED_NXDOMAIN
```

**Comment by netdevfighter** [comment]
@engelgabriel Yes, is a private network (RFC1918).  Is only internal reachable of the company.

**Comment by engelgabriel** [comment]
Ok, so I presume you can open the URL https://srv-rc-test.bs.kae.de.oneandone.net in your browsers and use Rocket.Chat with no problems?

**Comment by netdevfighter** [comment]
Yes, works fine without problems. 

![rocketchatrunning](https://cloud.githubusercontent.com/assets/4456399/14117435/f6b712d2-f5e4-11e5-8b2e-2940f4518903.png)

Let me know if you need more information.

For installation I've used the ansible script.

**Comment by rodrigok** [comment]
@netdevfighter: Your HTTPS is invalid, if your instance is not reachable via HTTP the app will not connect to your instance.

**Comment by engelgabriel** [comment]
The mobile and desktop app to not accept self signed certificates.

**Comment by engelgabriel** [comment]
That's why we try to access the URL without SSL after failing on HTTPS.

**Comment by netdevfighter** [comment]
Ok, but the error message is in this point incorrect. 

Is it not possible to irgnore this maybe with configuration field (checkbox)? 
Or can I import the ca-certificates of self signed certificates?

**Comment by netdevfighter** [comment]
The app works with http://srv-rc-test.bs.kae.de.oneandone.net:3000

The problem with the other url was the redirect to https from nginx.

**Comment by netdevfighter** [comment]
:+1:  An clear error message would have been desirable. ;)

Keep it up.

**Comment by engelgabriel** [comment]
Agreed, I have changed the error message. But the browser dont give a error code for invalid SSL certificate, it just return an error :( so there is not much we can work if.

**Comment by bott0r** [comment]
I had a similar problem, for me i have two domains DomainA.com hosts the RC instance and DomainB.com justs forwards to DomainA.com.

What works is, i have to directly input DomainA.com into the app in order to connect via the app. If i input DomainB.com it doesnt work. Of course both domains work in a regular browser.
